cmd/minishift/cmd/profile/profile_copy.go

```diff
@@ -0,0 +1,132 @@
+/*
+Copyright (C) 2017 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package profile
+
+import (
+	"fmt"
+	"os"
+
+	cmdUtil "github.com/minishift/minishift/cmd/minishift/cmd/util"
+	"github.com/minishift/minishift/cmd/minishift/state"
+	cmdState "github.com/minishift/minishift/cmd/minishift/state"
+	"github.com/minishift/minishift/pkg/minikube/constants"
+	"github.com/minishift/minishift/pkg/minishift/config"
+	minishiftConstants "github.com/minishift/minishift/pkg/minishift/constants"
+	"github.com/minishift/minishift/pkg/util/filehelper"
+	"github.com/minishift/minishift/pkg/util/os/atexit"
+	"github.com/spf13/cobra"
+)
+
+var (
+	profileCopyCmd = &cobra.Command{
+		Use:   "copy SRC_PROFILE_NAME NEW_PROFILE_NAME",
+		Short: "Copy configurations from an existing profile to a new profile.",
+		Long:  "Copy add-on and config configurations from an existing profile to a new profile.",
+		Run:   copyProfile,
+	}
+)
+
+func copyProfile(cmd *cobra.Command, args []string) {
+	validateCopyProfileCmd(args)
+	srcProfile := args[0]
+	newProfile := args[1]
+
+	if !cmdUtil.IsValidProfile(srcProfile) {
+		atexit.ExitWithMessage(1, fmt.Sprintf("Profile %s does not exist", srcProfile))
+	}
+
+	if cmdUtil.IsValidProfile(newProfile) {
+		atexit.ExitWithMessage(1, fmt.Sprintf("Profile '%s' already exists. You must provide a non-existant profile name", newProfile))
+	}
+	copySrcToNewProfile(srcProfile, newProfile)
+
+}
+
+func copySrcToNewProfile(srcProfile, newProfile string) {
+	srcConfFile := constants.GetProfileConfigFile(srcProfile)
+	sourceConfig, err := config.ReadViperConfig(srcConfFile)
+	if err != nil {
+		atexit.ExitWithMessage(1, err.Error())
+	}
+
+	newConfFile := constants.GetProfileConfigFile(newProfile)
+
+	// Check if the config file for the new profile is not empty
+	newConfig, err := config.ReadViperConfig(newConfFile)
```

Comment:
@LalatenduMohanty I am not sure which scenario it should be non-empty since we are already checking the the existence of  the `newprofile` and returning the error if present the doesn't it is known that new config is always going to to empty one?